akka-actor/src/main/scala/akka/actor/ActorCell.scala

```diff
@@ -680,25 +684,16 @@ private[akka] class ActorCell(
     case _                               =>
   }
 
-  final protected def clearActorCellFields(cell: ActorCell): Unit = {
-    cell.unstashAll()
-    if (!Reflect.lookupAndSetField(classOf[ActorCell], cell, "props", ActorCell.terminatedProps))
-      throw new IllegalArgumentException("ActorCell has no props field")
-  }
-
+  @silent
   final protected def clearActorFields(actorInstance: Actor, recreate: Boolean): Unit = {
-    setActorFields(actorInstance, context = null, self = if (recreate) self else system.deadLetters)
```

Comment:
add test from https://github.com/akka/akka/pull/27500/files#diff-3ba084826d373e8f2383f5a991c9db74R157

Reply:
I think we can remove `Reflect.lookupAndSetField` now. I didn't find any call to it in the Akka code base. Are we using it from other libraries (eg: akka-http)?

Reply:
Done